src/Interface/Modules/Visualization/InterfaceWithOsprayDialog.cc

```diff
@@ -61,15 +63,65 @@ InterfaceWithOsprayDialog::InterfaceWithOsprayDialog(const std::string& name, Mo
   addDoubleSpinBoxManager(defaultColorRDoubleSpinBox_, DefaultColorR);
   addDoubleSpinBoxManager(defaultColorGDoubleSpinBox_, DefaultColorG);
   addDoubleSpinBoxManager(defaultColorBDoubleSpinBox_, DefaultColorB);
+  addDoubleSpinBoxManager(defaultColorADoubleSpinBox_, DefaultColorA);
   addDoubleSpinBoxManager(backgroundColorRDoubleSpinBox_, BackgroundColorR);
   addDoubleSpinBoxManager(backgroundColorGDoubleSpinBox_, BackgroundColorG);
   addDoubleSpinBoxManager(backgroundColorBDoubleSpinBox_, BackgroundColorB);
+  addDoubleSpinBoxManager(streamlineRadiusDoubleSpinBox_, StreamlineRadius);
   addCheckBoxManager(showImageCheckBox_, ShowImageInWindow);
 
+  addDoubleSpinBoxManager(lightColorRDoubleSpinBox_, LightColorR);
+  addDoubleSpinBoxManager(lightColorGDoubleSpinBox_, LightColorG);
+  addDoubleSpinBoxManager(lightColorBDoubleSpinBox_, LightColorB);
+  addDoubleSpinBoxManager(lightIntensityDoubleSpinBox_, LightIntensity);
+  addCheckBoxManager(lightVisibleCheckBox_, LightVisible);
+  addComboBoxManager(lightTypeComboBox_, LightType);
+  addCheckBoxManager(automaticViewCheckBox_, AutoCameraView);
+  connect(automaticViewCheckBox_, SIGNAL(stateChanged(int)), this, SLOT(updateViewWidgets(int)));
+
+  connect(saveFileButton_, SIGNAL(clicked()), this, SLOT(saveFile()));
+  connect(fileNameLineEdit_, SIGNAL(editingFinished()), this, SLOT(pushFileNameToState()));
+  connect(fileNameLineEdit_, SIGNAL(returnPressed()), this, SLOT(pushFileNameToState()));
+
   createExecuteInteractivelyToggleAction();
 
   state_->connectSpecificStateChanged(Variables::Filename, [this]() { imageFilenameChanged(); });
   connect(this, SIGNAL(imageFilenameChanged()), this, SLOT(showImage()));
+  WidgetStyleMixin::tabStyle(tabWidget);
+
+  connect(closeImageWindowsPushButton_, SIGNAL(clicked()), this, SLOT(closeImageWindows()));
+
+  connect(zoomHorizontalSlider_, SIGNAL(sliderReleased()), this, SLOT(adjustZoom()));
+}
+
+void InterfaceWithOsprayDialog::adjustZoom()
+{
+  //qDebug() << __FUNCTION__ << zoomHorizontalSlider_->value() << zoomHorizontalSlider_->minimum() << zoomHorizontalSlider_->maximum();
+  Point camPos(state_->getValue(CameraPositionX).toDouble(), state_->getValue(CameraPositionY).toDouble(), state_->getValue(CameraPositionZ).toDouble());
+  //qDebug() << "camPos" << camPos.get_string().c_str();
+  Point camView(state_->getValue(CameraViewX).toDouble(), state_->getValue(CameraViewY).toDouble(), state_->getValue(CameraViewZ).toDouble());
+  //qDebug() << "camView" << camView.get_string().c_str();
+  auto a = camView - camPos;
+  //qDebug() << "distance: " << a.length();
+
+  auto zoomFactor = 2 * (zoomHorizontalSlider_->value() + 1.0) / (zoomHorizontalSlider_->maximum() + 1);
```

Comment:
There is a sign error somewhere around here.  The camera position flips sign when using this function.

Reply:
line 104 needs to be inverted

Reply:
Yeah i noticed that. Thanks, will fix.